x-pack/plugins/infra/public/pages/metrics/hosts/components/tabs/tabs.tsx

```diff
@@ -5,37 +5,71 @@
  * 2.0.
  */
 
-import React from 'react';
-import { EuiTabbedContent, EuiSpacer, type EuiTabbedContentTab } from '@elastic/eui';
+import React, { useRef, useState } from 'react';
+import { EuiTabs, EuiTab, EuiSpacer } from '@elastic/eui';
 import { i18n } from '@kbn/i18n';
 import { MetricsGrid } from './metrics/metrics_grid';
+import { AlertsTabContent } from './alerts';
+
+import { AlertsTabBadge } from './alerts_tab_badge';
+import { tabIds } from './config';
+
+const tabs = [
+  {
+    id: tabIds.METRICS,
+    name: i18n.translate('xpack.infra.hostsViewPage.tabs.metricsCharts.title', {
+      defaultMessage: 'Metrics',
+    }),
+    'data-test-subj': 'hostsView-tabs-metrics',
+  },
+  {
+    id: tabIds.ALERTS,
+    name: i18n.translate('xpack.infra.hostsViewPage.tabs.alerts.title', {
+      defaultMessage: 'Alerts',
+    }),
+    append: <AlertsTabBadge />,
+    'data-test-subj': 'hostsView_tab_alerts',
+  },
+];
+
+const initialRenderedTabsSet = new Set([tabs[0].id]);
+
+export const Tabs = () => {
+  // This map allow to keep track of which tabs content have been rendered the first time.
+  // We need it in order to load a tab content only if it gets clicked, and then keep it in the DOM for performance improvement.
+  const renderedTabsSet = useRef(initialRenderedTabsSet);
+
+  const [selectedTabId, setSelectedTabId] = useState(tabs[0].id);
+
+  const tabEntries = tabs.map((tab, index) => (
+    <EuiTab
+      {...tab}
+      key={index}
+      onClick={() => {
+        renderedTabsSet.current.add(tab.id); // On a tab click, mark the tab content as allowed to be rendered
+        setSelectedTabId(tab.id);
+      }}
```

Comment:
+1 on the addition of the current tab in the URL

Reply:
Love the idea, I just implemented this one since it was a quick win, thanks for the suggestion!

Reply:
Just realized this will cause re-rendering the whole sections and lose the performance gain we had with the tabs rendering. Also, it will cause an issue since moving from one tab to another will cause a data refetch even if not necessary.
I'll switch back to the previous implementation and leave this for another issue.